src/txn/txn.c

```diff
@@ -2948,6 +2948,10 @@ __wt_verbose_dump_txn(WT_SESSION_IMPL *session)
         if ((id = __wt_atomic_loadv64(&s->id)) == WT_TXN_NONE &&
           __wt_atomic_loadv64(&s->pinned_id) == WT_TXN_NONE)
             continue;
+
+        if (!F_ISSET(session->txn, WT_TXN_HAS_SNAPSHOT))
```

Comment:
I think we still need to dump other information, like pinned id, metadata pinned, and etc.

Reply:
@quchenhao 
Here is what I understand. Maybe my understanding is wrong, and I hope to get your guidance:
1. dump here is used to help us analyze transaction problems in extreme cases. If a transaction is released, I am not sure if this information is useful. When I simulated an exception offline, I found that there were a lot of transaction prints that had been released here.
2. dump here can help us understand "id pinned", "metadata pinned ID", etc. wt's transaction design is excellent, but it is difficult for beginners to understand the essence of transaction design, so for me, the bigger role here is to help beginner to understand the code with this information.

when the snapshot release, the "pinned id" and "metadata pinned id" will always be 0.
"pinned id" is the minimum transaction id in snapshot[]. For example, snapshot[]=[57632, 57633, 57634, 57635, 57636, 57637], 57632 is the checkpoint session's transaction id, then "pinned id"=57633, "metadata pinned id"=57632.  But the print here will always be 0, which will be inconsistent with the intent of the code.

Currently, I actually have another PR, which is based on the __wt_verbose_dump_txn function to help beginners understand the core design idea of transactions, because There are too many PR's that haven't been reviewed yet, so I have not committed it. By chance, I would like to take this opportunity to know if my above understanding is correct.

Reply:
```
when the snapshot release, the "pinned id" and "metadata pinned id" will always be 0.
```

This is not true. metadata pinned and pinned id will have a value even under read uncommitted isolation, which doesn't take a snapshot. This is why we should separate them. We should always dump the pinned ids but we can skip dumping the snapshot if we don't have one.

Reply:
Got it. Thanks for explaining, I have change the code, thanks.
@quchenhao